server/hotrod/src/main/scala/org/infinispan/server/hotrod/CacheDecodeContext.scala

```diff
@@ -261,18 +260,28 @@ class CacheDecodeContext(server: HotRodServer) extends ServerConstants with Log
     * Otherwise it's just considered number of seconds from
     * now and it's returned in milliseconds unit.
     */
-   protected def toMillis(lifespan: Int): Long = {
-      if (lifespan > SecondsInAMonth) {
-         val unixTimeExpiry = TimeUnit.SECONDS.toMillis(lifespan) - System.currentTimeMillis
+   protected def toMillis(param: ExpirationParam): Long = {
```

Comment:
@danberindei Or we could keep it in 2.2 (that needs to be backported unfortunately), and remove for 8.0 that for sure will need Hot Rod protocol change, wdyt?